control-plane/subcommand/gateway-resources/command_test.go

```diff
@@ -254,3 +268,130 @@ func TestRun(t *testing.T) {
+    spec:
+      gatewayClassName: consul-mesh-gateway
+`
+
+func TestRun_loadMesGatewayConfigs(t *testing.T) {
+	filename := createMeshGatewayConfigFile(t, validGWConfiguration)
+	// setup k8s client
+	s := runtime.NewScheme()
+	require.NoError(t, gwv1beta1.Install(s))
+	require.NoError(t, v1alpha1.AddToScheme(s))
+
+	client := fake.NewClientBuilder().WithScheme(s).Build()
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:                        ui,
+		k8sClient:                 client,
+		flagGatewayConfigLocation: filename,
+	}
+
+	err := cmd.loadGatewayConfigs()
+	require.NoError(t, err)
+	require.NotEmpty(t, cmd.gatewayConfig.GatewayClassConfigs)
+
+	// we only created one class config
+	classConfig := cmd.gatewayConfig.GatewayClassConfigs[0].DeepCopy()
+
+	// TODO: Add resources to the example yaml and test here once https://github.com/hashicorp/consul/pull/19725 merges
+	expectedClassConfig := v2beta1.GatewayClassConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "mesh.consul.hashicorp.com/v2beta1",
+			Kind:       "gatewayClassConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "consul-mesh-gateway",
+			Namespace: "namespace",
+		},
+		Spec:   meshv2beta1.GatewayClassConfig{},
+		Status: v2beta1.Status{},
+	}
+	require.Equal(t, expectedClassConfig.DeepCopy(), classConfig)
+}
+
+func TestRun_loadMesGatewayConfigsWithInvalidFile(t *testing.T) {
+	filename := createMeshGatewayConfigFile(t, invalidGWConfiguration)
+	// setup k8s client
+	s := runtime.NewScheme()
+	require.NoError(t, gwv1beta1.Install(s))
+	require.NoError(t, v1alpha1.AddToScheme(s))
+
+	client := fake.NewClientBuilder().WithScheme(s).Build()
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:                        ui,
+		k8sClient:                 client,
+		flagGatewayConfigLocation: filename,
+	}
+
+	err := cmd.loadGatewayConfigs()
+	require.Error(t, err)
+	require.Empty(t, cmd.gatewayConfig.GatewayClassConfigs)
+}
+
+func TestRun_loadGatewayConfigsWhenConfigFileDoesNotExist(t *testing.T) {
+	filename := "./consul/config/config.yaml"
+	s := runtime.NewScheme()
+	require.NoError(t, gwv1beta1.Install(s))
+	require.NoError(t, v1alpha1.AddToScheme(s))
+
+	client := fake.NewClientBuilder().WithScheme(s).Build()
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:                        ui,
+		k8sClient:                 client,
+		flagGatewayConfigLocation: filename,
+	}
+
+	err := cmd.loadGatewayConfigs()
+	require.NoError(t, err)
+	require.Empty(t, cmd.gatewayConfig.GatewayClassConfigs)
+	require.Contains(t, string(ui.ErrorWriter.Bytes()), "gateway configuration file not found, skipping gateway configuration")
+}
+
+func createMeshGatewayConfigFile(t *testing.T, fileContent string) string {
+	t.Helper()
+
+	// create a temp file to store configuration yaml
+	tmpdir := t.TempDir()
+	file, err := os.CreateTemp(tmpdir, "config.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = file.Write([]byte(fileContent))
+
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Might as well close the file here, even though it's for a test. There's also a useful alternative func for strings

```suggestion
	tmpdir := t.TempDir()
	file, err := os.CreateTemp(tmpdir, "config.yaml")
	if err != nil {
		t.Fatal(err)
	}
	defer file.Close()

	_, err = file.WriteString(fileContent)
	if err != nil {
		t.Fatal(err)
	}
```

Reply:
good catch!